src/plone/api/portal.py

```diff
@@ -277,36 +281,56 @@ def get_registry_record(name=None, interface=None):
     if interface is not None:
         records = registry.forInterface(interface)
         _marker = object()
-        if getattr(records, name, _marker) == _marker:
-            # Show all records on the interface.
-            records = [key for key in interface.names()]
-            raise InvalidParameterError(
-                "Cannot find a record with name '{0}' on interface {1}.\n"
-                'Did you mean?'
-                '{2}'.format(
-                    name,
-                    interface.__identifier__,
-                    '\n'.join(records)
-                )
+        if getattr(records, name, _marker) != _marker:
+            return registry['{0}.{1}'.format(interface.__identifier__, name)]
+
+        # Show all records on the interface.
+        records = [key for key in interface.names()]
+        msg = (
+            "Cannot find a record with name '{0}' on interface {1}.\n"
+            'Did you mean?'
+            '{2}'.format(
+                name,
+                interface.__identifier__,
+                '\n'.join(records)
             )
-        return registry['{0}.{1}'.format(interface.__identifier__, name)]
+        )
+        if default is MISSING:
+            raise InvalidParameterError(msg)
 
-    if name not in registry:
-        # Show all records that 'look like' name.
-        # We don't dump the whole list, because it 1500+ items.
-        records = [key for key in registry.records.keys() if name in key]
-        if records:
-            raise InvalidParameterError(
-                "Cannot find a record with name '{0}'.\n"
-                'Did you mean?:\n'
-                '{1}'.format(name, '\n'.join(records))
-            )
-        else:
-            raise InvalidParameterError(
-                "Cannot find a record with name '{0}'".format(name)
-            )
+        msg = '{0}\nReturning default {1}.'.format(
+            msg,
+            repr(default)
+        )
+        logger.warning(msg)
+        return default
+
+    if name in registry:
+        return registry[name]
+
+    # Show all records that 'look like' name.
+    # We don't dump the whole list, because it 1500+ items.
+    records = [key for key in registry.records.keys() if name in key]
+    if records:
+        msg = (
+            "Cannot find a record with name '{0}'.\n"
+            'Did you mean?:\n'
+            '{1}'.format(name, '\n'.join(records))
+        )
+    else:
+        msg = (
+            "Cannot find a record with name '{0}'".format(name)
+        )
 
-    return registry[name]
+    if default is MISSING:
+        raise InvalidParameterError(msg)
+
+    msg = '{0}\nReturning default {1}.'.format(
+        msg,
+        repr(default)
+    )
+    logger.warning(msg)
```

Comment:
Sorry @jaroel, I think I misunderstood your sentence.
Would you prefer to change the debug level to debug or to use warning.warn so that the log message will appear only once?